Pokedex/Models/PokemonResponse.swift

```diff
@@ -0,0 +1,62 @@
+//
+//  Pokemon.swift
+//  Pokedex
+//
+//  Created by krikaz on 6/9/20.
+//  Copyright © 2020 thewire. All rights reserved.
+//
+
+import Foundation
+
+private struct RawServerResponse: Decodable {
+    
+    var moves: [SingleMove]
+    var sprites: Sprite
+    var types: [SingleType]
+    var name: String
+
+    struct SingleMove: Decodable {
+        let move: MoveDetail
+    }
+    
+    struct MoveDetail: Decodable {
+        let name: String
+        let url: String
+    }
+    
+    struct Sprite: Decodable {
+        let front_default: String
+        let front_shiny: String
```

Comment:
Use camelCase in Swift. For JSON decoding, you can use [.keyDecodingStrategy = .convertFromSnakeCase](https://developer.apple.com/documentation/foundation/jsondecoder/keydecodingstrategy/convertfromsnakecase) when using `JSONDecoder`.

Reply:
I have followed the structure from this [stack answer ](https://stackoverflow.com/questions/44549310/how-to-decode-a-nested-json-struct-with-swift-decodable-protocol) so I didn't have to use CodingKeys.

Reply:
When you get the response from your callback, you can use a `JSONDecoder` to decode it. You can change the `.keyDecodingStrategy` like said above ☝️